externals/mocha/v5/src/index.Mocha.kt

```diff
@@ -0,0 +1,164 @@
+    @nativeGetter
+    operator fun get(index: String): Any?
+    @nativeSetter
+    operator fun set(index: String, value: Any)
+}
+external interface ITestCallbackContext {
+    fun skip(): ITestCallbackContext /* this */
+    fun timeout(ms: String): ITestCallbackContext /* this */
+    fun timeout(ms: Number): ITestCallbackContext /* this */
+    fun retries(n: Number): ITestCallbackContext /* this */
+    fun slow(ms: Number): ITestCallbackContext /* this */
+    @nativeGetter
+    operator fun get(index: String): Any?
+    @nativeSetter
+    operator fun set(index: String, value: Any)
+}
+external interface IRunnable {
+    var title: String
+    var fn: Function<*>
+    var async: Boolean
+    var sync: Boolean
+    var timedOut: Boolean
+    fun timeout(n: String): IRunnable /* this */
+    fun timeout(n: Number): IRunnable /* this */
+    var duration: Number? get() = definedExternally; set(value) = definedExternally
+}
+external interface ISuite {
+    var parent: ISuite
+    var title: String
+    fun fullTitle(): String
+}
+external interface ITest : IRunnable {
+    var parent: ISuite
+    var pending: Boolean
+    var state: dynamic /* String /* "failed" */ | String /* "passed" */ | Nothing? */
+    fun fullTitle(): String
+}
+external interface IBeforeAndAfterContext : IHookCallbackContext {
+    var currentTest: ITest
+}
+external interface IStats {
+    var suites: Number
+    var tests: Number
+    var passes: Number
+    var pending: Number
+    var failures: Number
+    var start: Date? get() = definedExternally; set(value) = definedExternally
+    var end: Date? get() = definedExternally; set(value) = definedExternally
+    var duration: Date? get() = definedExternally; set(value) = definedExternally
+}
+external interface IRunner {
+    var stats: IStats? get() = definedExternally; set(value) = definedExternally
+    var started: Boolean
+    var suite: ISuite
+    var total: Number
+    var failures: Number
+    var grep: (re: String, invert: Boolean) -> IRunner /* this */
+    var grepTotal: (suite: ISuite) -> Number
+    var globals: (arr: Array<String>) -> dynamic /* Array<String> | IRunner */
+    var abort: () -> IRunner /* this */
+    var run: (fn: ((failures: Number) -> Unit)? /*= null*/) -> IRunner /* this */
+}
+external interface IContextDefinition {
+    @nativeInvoke
+    operator fun invoke(description: String, callback: (`this`: ISuiteCallbackContext) -> Unit): ISuite
+    fun only(description: String, callback: (`this`: ISuiteCallbackContext) -> Unit): ISuite
+    fun skip(description: String, callback: (`this`: ISuiteCallbackContext) -> Unit)
+    fun timeout(ms: String)
+    fun timeout(ms: Number)
+}
+external interface ITestDefinition {
+    @nativeInvoke
+    operator fun invoke(expectation: String, callback: ((`this`: ITestCallbackContext, done: (error: Any? /*= null*/) -> Unit) -> dynamic /* Unit | PromiseLike<Any> */)? = definedExternally /* null */): ITest
```

Comment:
I am pretty sure the `callback` is going to be a bit troublesome.
Mocha expects a function with either 0 or 1 argument and treats the test as either synchronous or asynchronous accordingly. At the moment we cannot describe a function with `this` of a specific type (extension functions are translated to functions with an additional first argument `$receiver`). So the `callback` is going to be a function with 2 arguments. I am not sure how Mocha would treat that. Could you please check?